"error initializing LibGit2 module" when CA certificates not found

Using openSUSE Leap 42.1: 
24 hours ago a git pull followed by a `make && make testall` ran fine. This morning I had an error in make:

```
...
patching file tests/online/badssl.c
CMake Error: The current CMakeCache.txt directory /home/colin/Downloads/julia6/deps/scratch/libgit2-211e117a0590583a720c53172406f34186c543bd/CMakeCache.txt is different than the directory /home/colin/Downloads/julia/deps/scratch/libgit2-211e117a0590583a720c53172406f34186c543bd where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
CMake Error: The source "/home/colin/Downloads/julia6/deps/srccache/libgit2-211e117a0590583a720c53172406f34186c543bd/CMakeLists.txt" does not match the source "/home/colin/Downloads/julia/deps/srccache/libgit2-211e117a0590583a720c53172406f34186c543bd/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.
/home/colin/Downloads/julia6/deps/libgit2.mk:66: recipe for target 'scratch/libgit2-211e117a0590583a720c53172406f34186c543bd/build-configured' failed
make[1]: *** [scratch/libgit2-211e117a0590583a720c53172406f34186c543bd/build-configured] Error 1
Makefile:81: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
```

I thought I had fixed this with `make -C deps distclean-libgit2 distclean-mbedtls distclean-libssh2` which allowed `make` to run to the end. However now Julia 0.6 fails to launch with:

```
> julia6
fatal: error thrown and no exception handler available.
Base.InitError(mod=:LibGit2, error=ErrorException("error initializing LibGit2 module"))
rec_backtrace at /home/colin/Downloads/julia6/src/stackwalk.c:84
record_backtrace at /home/colin/Downloads/julia6/src/task.c:238
jl_throw at /home/colin/Downloads/julia6/src/task.c:560
__init__ at ./libgit2/libgit2.jl:539
unknown function (ip: 0x7f60d03e3e08)
jl_call_method_internal at /home/colin/Downloads/julia6/src/julia_internal.h:218 [inlined]
jl_apply_generic at /home/colin/Downloads/julia6/src/gf.c:1852
jl_apply at /home/colin/Downloads/julia6/src/julia.h:1376 [inlined]
jl_module_run_initializer at /home/colin/Downloads/julia6/src/toplevel.c:83
_julia_init at /home/colin/Downloads/julia6/src/init.c:702
julia_init at /home/colin/Downloads/julia6/src/task.c:289
unknown function (ip: 0x4014cd)
__libc_start_main at /lib64/libc.so.6 (unknown line)
unknown function (ip: 0x40151f)

```

Prebuilt Julia 0.5 works fine.

One curiosity is that if I do a `find ./ -name libgit2.so` from julia main directory I find eight different files matched, some clearly symlinks, but others may be different residual versions causing confusion.
```
> ldd usr/lib/libgit2.so
	linux-vdso.so.1 (0x00007ffeced63000)
	libcurl.so.4 => /home/colin/Downloads/julia6/usr/lib/libcurl.so.4 (0x00007f5926bf8000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f59269b2000)
	libmbedtls.so.10 => /home/colin/Downloads/julia6/usr/lib/libmbedtls.so.10 (0x00007f592678b000)
	libmbedx509.so.0 => /home/colin/Downloads/julia6/usr/lib/libmbedx509.so.0 (0x00007f5926577000)
	libmbedcrypto.so.0 => /home/colin/Downloads/julia6/usr/lib/libmbedcrypto.so.0 (0x00007f5926323000)
	libssh2.so.1 => /home/colin/Downloads/julia6/usr/lib/libssh2.so.1 (0x00007f59260f0000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f5925ee8000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f5925cca000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f5925922000)
	/lib64/ld-linux-x86-64.so.2 (0x00005572745ee000)

```
Then:
`make cleanall` is successful, but make finishes with:
```
...
configure: creating ./config.status
config.status: creating libuv.pc
config.status: creating Makefile
config.status: executing depfiles commands
config.status: executing libtool commands
make[2]: *** No rule to make target '/home/colin/Downloads/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/fs-poll.c', needed by 'src/libuv_la-fs-poll.lo'.  Stop.
/home/colin/Downloads/julia6/deps/libuv.mk:37: recipe for target 'scratch/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/build-compiled' failed
make[1]: *** [scratch/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/build-compiled] Error 2
Makefile:81: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2

```

After `make -C deps distclean-libuv` make still failing. 
```
> make
 cd /home/colin/Downloads/julia/deps/srccache/libunwind-1.1-julia2 && /bin/sh /home/colin/Downloads/julia/deps/srccache/libunwind-1.1-julia2/config/missing automake-1.15 --gnu Makefile
/bin/sh: line 10: cd: /home/colin/Downloads/julia/deps/srccache/libunwind-1.1-julia2: No such file or directory
Makefile:477: recipe for target '/home/colin/Downloads/julia/deps/srccache/libunwind-1.1-julia2/Makefile.in' failed
make[2]: *** [/home/colin/Downloads/julia/deps/srccache/libunwind-1.1-julia2/Makefile.in] Error 1
/home/colin/Downloads/julia6/deps/unwind.mk:31: recipe for target '/home/colin/Downloads/julia6/usr-staging/libunwind-1.1-julia2.tgz' failed
make[1]: *** [/home/colin/Downloads/julia6/usr-staging/libunwind-1.1-julia2.tgz] Error 2
Makefile:81: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2

```

I just hit this on power:
```
    JULIA test/all
fatal: error thrown and no exception handler available.
Base.InitError(mod=:LibGit2, error=ErrorException("error initializing LibGit2 module"))
rec_backtrace at /home/valentin/juliatest/src/stackwalk.c:84
record_backtrace at /home/valentin/juliatest/src/task.c:238
jl_throw at /home/valentin/juliatest/src/task.c:560
__init__ at ./libgit2/libgit2.jl:539
unknown function (ip: 0x3fffa68cf1bf)
jl_call_method_internal at /home/valentin/juliatest/src/julia_internal.h:239 [inlined]
jl_apply_generic at /home/valentin/juliatest/src/gf.c:1852
jl_apply at /home/valentin/juliatest/src/julia.h:1376 [inlined]
jl_module_run_initializer at /home/valentin/juliatest/src/toplevel.c:83
_julia_init at /home/valentin/juliatest/src/init.c:702
julia_init at /home/valentin/juliatest/src/task.c:289
main at /home/valentin/juliatest/ui/repl.c:247
unknown function (ip: 0x3fffb226457f)
__libc_start_main at /lib64/power8/libc.so.6 (unknown line)
make: *** [all] Error 1
make: Leaving directory `/home/valentin/juliatest/test'
```

I'm getting this problem on Ubuntu again.

```
fengyang@serval ~/julia> make test
    JULIA test/all
fatal: error thrown and no exception handler available.
Base.InitError(mod=:LibGit2, error=ErrorException("error initializing LibGit2 module"))
rec_backtrace at /home/fengyang/julia/src/stackwalk.c:84
record_backtrace at /home/fengyang/julia/src/task.c:238
jl_throw at /home/fengyang/julia/src/task.c:560
__init__ at ./libgit2/libgit2.jl:558
unknown function (ip: 0x7fa1f2dc6c68)
jl_call_method_internal at /home/fengyang/julia/src/julia_internal.h:239 [inlined]
jl_apply_generic at /home/fengyang/julia/src/gf.c:1861
jl_apply at /home/fengyang/julia/src/julia.h:1378 [inlined]
jl_module_run_initializer at /home/fengyang/julia/src/toplevel.c:83
_julia_init at /home/fengyang/julia/src/init.c:696
julia_init at /home/fengyang/julia/src/task.c:289
main at /home/fengyang/julia/ui/repl.c:247
__libc_start_main at /build/glibc-GKVZIf/glibc-2.23/csu/../csu/libc-start.c:291
unknown function (ip: 0x401438)
Makefile:19: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:556: recipe for target 'test' failed
make: *** [test] Error 2
```

Do I need to change certs again?

--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -47,6 +47,10 @@ OPENSSL_DIR=$(shell openssl version -d | cut -d '"' -f 2)
 ifeq ($(shell [ -e $(OPENSSL_DIR)/cert.pem ] && echo exists),exists)
 CERTFILE=$(OPENSSL_DIR)/cert.pem
 endif
+# This certfile location observed on Ubuntu 16.04
+ifeq ($(shell [ -e $(OPENSSL_DIR)/certs/ca-certificates.crt ] && echo exists),exists)
+CERTFILE=$(OPENSSL_DIR)/certs/ca-certificates.crt
+endif
 # This certfile location observed on Debian 7
 ifeq ($(shell [ -e $(OPENSSL_DIR)/certs/ca.pem ] && echo exists),exists)
 CERTFILE=$(OPENSSL_DIR)/certs/ca.pem

